src/cryptography/x509/base.py

```diff
@@ -1169,5 +1184,69 @@ def build(self, backend: typing.Any = None) -> RevokedCertificate:
         )
 
 
+class Profile(utils.Enum):
+    RFC5280 = 0
+    WebPKI = 1
+
+
+class PolicyBuilder:
+    def __init__(
+        self,
+        *,
+        name: typing.Optional[str] = None,
+        time: typing.Optional[datetime.datetime] = None,
+        profile: Profile = Profile.WebPKI,
+    ):
+        self._name = name
+        self._time = time
+        self._profile = profile
+
+    def name(self, new_name: str) -> PolicyBuilder:
+        """
+        Sets the expected certificate subject.
+        """
+        self._name = new_name
+        return self
```

Comment:
Our builders do not use mutation.

Reply:
Okay, can change. Out of curiosity: why do you prefer non-mutating builders?

Reply:
Because it means builders can be reused in multiple places. You can create a builder with N fields filled in, store it somewhere, and then fill in the final field repeatedly.

Unlike in Rust, where we can use the type system to ensure no confusing sharing, in Python we don't have that, so immutability is better.

Reply:
Makes sense, thanks for explaining -- I hadn't considered that someone would store a builder.

Reply:
f21a4e98a